playbooks/common/openshift-cluster/upgrades/v3_1_to_v3_2/pre.yml

```diff
@@ -110,8 +110,8 @@
   # are modified to use the correct image tag.  However, this can trigger a
   # docker restart if new configuration is laid down which would immediately
   # pull the latest image and defeat the purpose of these tasks.
-  - openshift_cli
-  pre_tasks:
+  - openshift_docker
```

Comment:
Hmm, am I following you correctly:

All of this would still take place in pre.yml. We already have a play for first master, all masters, all nodes, and all nodes + all masters. 

I'm assuming this would be adding a second, new play for just the first master, it would immediately follow the existing one for first master, it would apply docker openshift role and query new versions.

Then add a second new play for all hosts minus first master, which follows all existing verify steps, includes openshift_docker role, and then uses the version from the master.

So we'd still be modifying docker related things in pre.yml, just later in the process, and we'd gain a little more confidence that everybody will be trying to use the same version?